clients/web/test/specRunner.js

```diff
@@ -176,6 +176,16 @@ page.onLoadFinished = function (status) {
         page.evaluate(function () {
             if (window.girderTest) {
                 girderTest.promise.done(function () {
+                    // Allow Jasmine to compare RegExp using toEqual, toHaveBeenCalledWith, etc.
+                    jasmine.getEnv().addEqualityTester(function (a, b) {
```

Comment:
`toHaveBeenCalledWith` is not a general matcher, it's specific to asserting that a spied method was called at some point in time with the given argument(s) (in any parameter position). To do so, it delegates to [`jasmine.Env.prototype.equals_`](https://github.com/jasmine/jasmine/blob/v1.3.1/src/core/Env.js#L228), which already supports the comparison of two actual regexes. This method also has lots of "equality" testers that perform matching in all sorts of non-strict ways (this is the same method used by the `toEqual` matcher). Since `toHaveBeenCalledWith` is already doing loose `toEqual`-style comparison, instead of strict (using `===`) `toBe`-style comparison, I'd call it a feature that you can now test spied arguments against regexes. If a user ever wanted really strict comparison, they shouldn't use `toHaveBeenCalledWith` in the first place, regardless of whether this additional equality tester is present.

However, this does highlight a small bug in how this custom equality tester was added: it should also ensure that the other argument is a string, or else the [`RegExp.prototype.test`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) could be called with an invalid argument. Once that bug is fixed, comparison of two regexes will continue to work properly in all cases.